streamer/src/sendmmsg.rs

```diff
@@ -4,157 +4,155 @@ use std::io;
 use std::net::{SocketAddr, UdpSocket};
 
 #[cfg(not(target_os = "linux"))]
-pub fn send_mmsg(sock: &UdpSocket, packets: &[(&Vec<u8>, &SocketAddr)]) -> io::Result<usize> {
-    let count = packets.len();
+pub fn batch_send(
+    sock: &UdpSocket,
+    packets: &[(&[u8], &SocketAddr)],
+) -> Result<(), (io::Error, usize)> {
+    let mut total_sent = 0;
+    let mut erropt = None;
     for (p, a) in packets {
-        sock.send_to(p, *a)?;
+        if let Err(e) = sock.send_to(p, *a) {
+            if erropt.is_none() {
+                erropt = Some(e);
+            }
+        } else {
+            total_sent += 1;
+        }
     }
 
-    Ok(count)
+    if let Some(err) = erropt {
+        Err((err, total_sent))
+    } else {
+        Ok(())
+    }
 }
 
 #[cfg(target_os = "linux")]
-use libc::{iovec, mmsghdr, sockaddr_in, sockaddr_in6};
+use libc::{iovec, mmsghdr, sockaddr_in, sockaddr_in6, sockaddr_storage};
 
 #[cfg(target_os = "linux")]
 fn mmsghdr_for_packet(
     packet: &[u8],
     dest: &SocketAddr,
     index: usize,
```

Comment:
This `index:usize` argument is redundant. You can remove this argument and instead add:
```
let index = hdrs.len();
```
before `hdrs.push(...)` line.

Reply:
done